Source/Core/Core/IOS/USB/Emulated/Microphone.cpp

```diff
@@ -0,0 +1,370 @@
+// Copyright 2025 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "Core/IOS/USB/Emulated/Microphone.h"
+
+#include <span>
+
+#ifdef HAVE_CUBEB
+#include <cubeb/cubeb.h>
+
+#include "AudioCommon/CubebUtils.h"
+#endif
+
+#include "Common/Logging/Log.h"
+#include "Common/MathUtil.h"
+#include "Common/Swap.h"
+#include "Core/Config/MainSettings.h"
+#include "Core/Core.h"
+#include "Core/IOS/USB/Emulated/WiiSpeak.h"
+#include "Core/System.h"
+
+#ifdef _WIN32
+#include <Objbase.h>
+#endif
+
+#ifdef ANDROID
+#include "jni/AndroidCommon/IDCache.h"
+#endif
+
+namespace IOS::HLE::USB
+{
+Microphone::Microphone(const WiiSpeakState& sampler) : m_sampler(sampler)
+{
+  StreamInit();
+}
+
+Microphone::~Microphone()
+{
+  StreamTerminate();
+}
+
+#ifndef HAVE_CUBEB
+void Microphone::StreamInit()
+{
+}
+
+void Microphone::StreamStart([[maybe_unused]] u32 sampling_rate)
+{
+}
+
+void Microphone::StreamStop()
+{
+}
+
+void Microphone::StreamTerminate()
+{
+}
+#else
+void Microphone::StreamInit()
+{
+  if (!m_worker.Execute([this] { m_cubeb_ctx = CubebUtils::GetContext(); }))
+  {
+    ERROR_LOG_FMT(IOS_USB, "Failed to init Wii Speak stream");
+    return;
+  }
+
+  // TODO: Not here but rather inside the WiiSpeak device if possible?
+  StreamStart(m_sampler.DEFAULT_SAMPLING_RATE);
+}
+
+void Microphone::StreamTerminate()
+{
+  StreamStop();
+
+  if (m_cubeb_ctx)
+    m_worker.Execute([this] { m_cubeb_ctx.reset(); });
+}
+
+static void state_callback(cubeb_stream* stream, void* user_data, cubeb_state state)
+{
+}
+
+void Microphone::StreamStart(u32 sampling_rate)
+{
+  if (!m_cubeb_ctx)
+    return;
+
+  m_worker.Execute([this, sampling_rate] {
+#ifdef ANDROID
+    JNIEnv* env = IDCache::GetEnvForThread();
+    if (jboolean result = env->CallStaticBooleanMethod(
+            IDCache::GetPermissionHandlerClass(),
+            IDCache::GetPermissionHandlerHasRecordAudioPermission(), nullptr);
+        result == JNI_FALSE)
+    {
+      env->CallStaticVoidMethod(IDCache::GetPermissionHandlerClass(),
+                                IDCache::GetPermissionHandlerRequestRecordAudioPermission(),
+                                nullptr);
+    }
+#endif
+
+    cubeb_stream_params params{};
+    params.format = CUBEB_SAMPLE_S16LE;
+    params.rate = sampling_rate;
+    params.channels = 1;
+    params.layout = CUBEB_LAYOUT_MONO;
+
+    u32 minimum_latency;
+    if (cubeb_get_min_latency(m_cubeb_ctx.get(), &params, &minimum_latency) != CUBEB_OK)
+    {
+      WARN_LOG_FMT(IOS_USB, "Error getting minimum latency");
```

Comment:
`minimum_latency` should probably at least be initialized if this is going to just be a `WARN_LOG`.